### ISSUE TITLE
[PAGE ISSUE]: 'The Dart type system' Should the "Generic type assignment" section mention potential runtime failures?

### ISSUE THREAD

**Issue by jamesderlin** [dart-lang/site-www#3920]
### Page URL

https://dart.dev/guides/language/type-system#generic-type-assignment

### Page source

https://github.com/dart-lang/site-www/tree/main/src/_guides/language/type-system.md

### Describe the problem

Should the "Generic type assignment" section (https://dart.dev/guides/language/type-system#generic-type-assignment) mention that treating `Generic<Subtype>` as a subtype of `Generic<Supertype>` could introduce potential runtime errors?  For example:

```dart
abstract class Animal {}
class Cat extends Animal {}
class Dog extends Animal {}

void addAnimal(List<Animal> animals) {
  animals.add(Dog()); // Looks safe, but fails at runtime.
}

void main() {
  var myCats = <Cat>[];
  addAnimal(myCats);
}
```

**Comment by parlough** [comment]
Good idea. We do mention this in [Fixing common type problems](https://dart.dev/guides/language/sound-problems#runtime-errors) but you’re right it could be worth mentioning as a warning here. 

Thanks for the suggestion :)

**Comment by jamesderlin** [comment]
I don't seem to see this specific kind of problem mentioned on that page.  I think the case of `Generic<Subtype>` being substitutable for `Generic<Supertype>` is worth mentioning explicitly since, unlike the cases that use `dynamic` or where there are explicit casts, it *looks* safe (particularly from the perspective of `addAnimal`), especially since it's implicitly allowed without any static analysis warnings.
